src/storage/async_storage/mod.rs

```diff
@@ -0,0 +1,92 @@
+use crate::errors::Result;
+use crate::operation::Operation;
+use crate::server::VersionId;
+use crate::storage::TaskMap;
+use async_trait::async_trait;
+use uuid::Uuid;
+
+pub(crate) mod adapter;
+
+/// An async Storage service, allowing storage operations.
+#[async_trait]
+pub(crate) trait AsyncStorage: Send + Sync {
```

Comment:
Part of the reason to do this now is to make the public trait -- which users might implement -- async, as a breaking change along with the other changes discussed in #599 like making `Replica::sync` async. So,
 * This trait should be named `Storage` (and similarly `StorageTxn`), with the blocking trait renamed to `BlockingStorage`.
 * It should be pub.
 * The existing crate (now `BlockingStorage`) should be `pub(crate)` so it's not user-visible
 * Since it will be the _only_ such trait in the future (when support for the blocking trait is dropped), it should have pride of place in `src/storage/mod.rs` where the existing trait is.

The overall effect from an external (`pub`) perspective is that all of the methods in the `Storage` traits became async.